curator-framework/src/main/java/org/apache/curator/framework/imps/EnsembleTracker.java

```diff
@@ -166,12 +166,26 @@ public static String configToConnectionString(QuorumVerifier data) throws Except
         StringBuilder sb = new StringBuilder();
         for ( QuorumPeer.QuorumServer server : data.getAllMembers().values() )
         {
+            if ( server.clientAddr == null )
+            {
+                // Invalid client address configuration in zoo.cfg
+                continue;
+            }
             if ( sb.length() != 0 )
             {
                 sb.append(",");
             }
-            InetSocketAddress address = Objects.firstNonNull(server.clientAddr, server.addr);
-            sb.append(address.getAddress().getHostAddress()).append(":").append(address.getPort());
+            InetAddress wildcardAddress = new InetSocketAddress(0).getAddress();
```

Comment:
I think you can just use server.addr which is guaranteed not null, and then use getHostAddress to get the hostAddress.